charts/azuremonitor-containers/templates/omsagent-daemonset-windows.yaml

```diff
@@ -25,8 +25,13 @@ spec:
     dockerProviderVersion: {{ .Values.omsagent.image.dockerProviderVersion }}
     schema-versions: "v1"
   spec:
+{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion }}
```

Comment:
i am not sure about this change. we should ideally do a 'or' so we work on both, as  this change requires customer workloads to be updated and most people dont. and also in on prem envs, this is highly customizable.

Reply:
> i am not sure about this change. we should ideally do a 'or' so we work on both, as this change requires customer workloads to be updated and most people dont. and also in on prem envs, this is highly customizable.

> No, this doesnt require customer to upgrade their k8s cluster since we handle pre 1.14 and higher.

Reply:
i am not saying this agent change requires customers to upgrade k8s. This label change requires customers workload to be updated (just like our agent) ,and most customers, just add this label back instead of updating their side, as node labels are templated in k8s installation.

Reply:
> i am not saying this agent change requires customers to upgrade k8s. This label change requires customers workload to be updated (just like our agent) ,and most customers, just add this label back instead of updating their side, as node labels are templated in k8s installation.

> Since we are using nodeselector, we cant express  `or` condition and nodeSelector only supports `and`.  If the customer doesnt upgraded k8s version but just adds the  `kubernetes.io/os` then this should work as long as they dont remove beta.kubernetes.io label.